internal/provider/alert_resource_unit_test.go

```diff
@@ -0,0 +1,253 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func Test_ValidateConditions_LengthLessThanOne(t *testing.T) {
+
+	model := alertResourceModel{
+		Conditions: []alertConditionModel{},
+	}
+	result := model.validateConditions()
+
+	expected := []diag.DiagnosticWithPath{
+		diag.NewAttributeErrorDiagnostic(
+			path.Root("conditions"),
+			"Invalid number of alerting conditions.",
+			"Number of alerting conditions must be between 1 and 5."),
+	}
+
+	if len(expected) != len(result) {
+		t.Fatalf("expected %v diagnosticErrors", len(expected))
+	}
+
+	for i := range expected {
+		r, _ := result[i].(diag.DiagnosticWithPath)
+		if !expected[i].Equal(r) {
+			t.Errorf("expected(%v, %v, %v) unexpected(%v, %v, %v) ",
+				expected[i].Path(), expected[i].Summary(), expected[i].Detail(),
+				r.Path(), r.Summary(), r.Detail())
+		}
+	}
```

Comment:
This can be simpler. You just need to return the correct type from `validateConditions`.

```suggestion
	expected := diag.Diagnostics{
		diag.NewAttributeErrorDiagnostic(
			path.Root("conditions"),
			"Invalid number of alerting conditions.",
			"Number of alerting conditions must be between 1 and 5.",
		),
	}

	if !result.Equal(expected) {
		t.Errorf("expected %v but got %v", expected, result)
	}
```

Reply:
Same thing for the rest of the cases below.

Reply:
this is what i was trying to do yesterday, but got all turned around with Diagnostic vs Diagnostics. >_<